UnityProject/Assets/Scripts/Objects/Consoles/ArtifactConsole.cs

```diff
@@ -0,0 +1,116 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Items.Science;
+using Objects.Research;
+using Mirror;
+using Shared.Systems.ObjectConnection;
+
+namespace Objects.Research
+{
+	public class ArtifactConsole : NetworkBehaviour, ICheckedInteractable<HandApply>, IMultitoolSlaveable
+	{
+		private RegisterObject registerObject;
+		public ItemStorage itemStorage;
+
+		public Artifact connectedArtifact;
+		public ArtifactDataDisk dataDisk;
+
+		[SyncVar] public ArtifactData inputData = new ArtifactData();
+
+		public delegate void StateChange();
+		public static event StateChange stateChange;
```

Comment:
Can you use an action instead of the delegate and just invoke it later in GUI update method?